doc/dev/debug_e2e_flake.md

```diff
@@ -0,0 +1,35 @@
+# Debug E2E Test Flake
+
+E2E flakes will cost more and more effort and bring more and more trouble as time passes by.
+It is better to solve them as early as possible.
+
+Rules when a flake happened:
+
+- Even though no engineer wants to, we need to accept that the code is buggy.
+- Do not hide the failure.
+  If test flake happened, don't just rerun the test.
+  Comment or create an issue to track the problem.
+- Figure out why the failure happened.
+  The end goal should be either a code fix or some docs explaining the limitatons.
+
+## How to debug
+
+Figuring out the root cause of failures is usually easier and less time-consuming than it's assumed to.
+
+Here is one process to help start the debugging process:
+
+- Have a automated script to reproduce the test consistently. For example, a script:
+  ```shell
+  for ((n=0;n<30;n++))
+  do
+  echo "${n} round..."
+  ./e2e.test --kubeconfig=${HOME}/.kube/config \
```

Comment:
can e2e test being run locally on minikube?

Reply:
It is better to use the same configuration as where the failure happens. That say, if it failed on jenkins, use the jenkins cluster or create another same cluster.

Reply:
e2e test is run on jenkin instance but the kube part is run on gke right?

Reply:
right